File: src/Backend/Inspections.Core/Domain/CheckListAggregate/CheckListItem.cs
using Inspections.Shared;

namespace Inspections.Core.Domain.CheckListAggregate;

public class CheckListItem : Entity<int>
{
    public int CheckListId { get; set; }
    public string Text { get; set; } = default!;
    public CheckValue Checked { get; set; }
    public bool Editable { get; set; }
    public bool Required { get; set; }
    public string? Remarks { get; set; }
    public bool Touched { get; set; }

    private CheckListItem() { } //Required by EF

    public CheckListItem(int checkListId, string text, CheckValue @checked, bool editable, bool required, string? remarks, bool touched = false)
    {
        CheckListId = checkListId;
        Text = text;
        Checked = @checked;
        Editable = editable;
        Required = required;
        Remarks = remarks;
        Touched = touched;
    }
}

File: src/ClientApp/types/CheckLists/Models/CheckListItem.ts
import { CheckListParam } from './CheckListParam'
import { CheckValue } from '~/services/api'

export interface CheckListItem {
  id: number
  checkListId: number
  text: string
  checked: CheckValue
  editable: boolean
  required: boolean
  remarks: string
  textParams: CheckListParam[]
}
